docs/concepts/configuration/constrain-pods-nodes.md

```diff
@@ -0,0 +1,203 @@
+---
+assignees:
+- davidopp
+- kevin-wangzefeng
+title: Constraining Pods to Run on Certain Nodes
+---
+
+{% capture overview %}
+
+You can constrain a Pod to be able to run only on particular
+[nodes](/docs/admin/node/) or to prefer to run on particular nodes. There are
+several ways to do this, and they all use
+[label selectors](/docs/user-guide/labels/).
+Generally such constraints are unnecessary, as the scheduler automatically does
+a reasonable placement. For example, the scheduler spreads your Pods across nodes
+and refrains from placing Pods on nodes that have insufficient free resources.
+But there are some circumstances where you might want more control of where a Pod
+lands. For example, you might want to ensure that a Pod ends up on a machine with
+an SSD attached to it. Or if two different services communicate with each other, you
+might want to locate the Pods for both services in the same availability zone.
+
+{% endcapture %}
+
+{% capture body %}
+
+## Using the nodeSelector field
+
+The simplest constraint is the `nodeSelector` field of a PodSpec. It
+specifies a map of key-value pairs. For the Pod to be eligible to run on a node,
+the node must have each of the specified key-value pairs as labels. It can have
+additional labels as well. The most common case is to use a single key-value pair.
+
+Here's a Pod configuration file that has a `nodeSelector` field:
+
+{% include code.html language="yaml" file="pod-node-selector.yaml" ghlink="/docs/concepts/configuration/pod-node-selector.yaml" %}
+
+In the configuration file you can see that the Pod will get scheduled on a node that
+has the label `disktype: ssd`.
+
+## Affinity
+
+The affinity feature greatly expands the types of constraints you can
+express. As of Kubernetes version 1.5, affinity is an alpha feature.
+The key enhancements are:
+
+* The language is more expressive than the `nodeSelector` field.
```

Comment:
I think the example that comes below is probably enough. But if you can think of a one- or two-line example, I could put it in this bullet point.